python_modules/dagster/dagster/core/asset_defs/assets_job.py

```diff
@@ -101,26 +103,79 @@ def build_job_partitions_from_assets(
     assets: Sequence[AssetsDefinition],
 ) -> Optional[PartitionedConfig]:
     assets_with_partitions_defs = [assets_def for assets_def in assets if assets_def.partitions_def]
-    if assets_with_partitions_defs:
-        for assets_def in assets_with_partitions_defs:
-            if assets_def.partitions_def != assets_with_partitions_defs[0].partitions_def:
-                first_asset_key = next(iter(assets_def.asset_keys)).to_string()
-                second_asset_key = next(iter(assets_with_partitions_defs[0].asset_keys)).to_string()
-                raise DagsterInvalidDefinitionError(
-                    "When an assets job contains multiple partitions assets, they must have the "
-                    f"same partitions definitions, but asset '{first_asset_key}' and asset "
-                    f"'{second_asset_key}' have different partitions definitions. "
-                )
 
-        return PartitionedConfig(
-            partitions_def=cast(
-                PartitionsDefinition, assets_with_partitions_defs[0].partitions_def
-            ),
-            run_config_for_partition_fn=lambda p: {},
-        )
-    else:
+    if len(assets_with_partitions_defs) == 0:
         return None
 
+    for assets_def in assets_with_partitions_defs:
+        if assets_def.partitions_def != assets_with_partitions_defs[0].partitions_def:
+            first_asset_key = next(iter(assets_def.asset_keys)).to_string()
+            second_asset_key = next(iter(assets_with_partitions_defs[0].asset_keys)).to_string()
```

Comment:
nit: set `assets_with_partitions_defs[0]` in to a local var to clarify this behavior